src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -1651,6 +1656,19 @@ parser::parse_and_visit_function_parameters(
     });
     return function_parameter_parse_result::missing_parameters_ignore_body;
 
+    // function async f() {}  // Invalid. Should be async function f() {}
+  case token_type::identifier:
+    if (name->string_view() ==
+        u8"async"_sv) {  // In this context async is of type identifier and not
+                         // kw_async so we need to look at the name.
+      this->diag_reporter_->report(
+          diag_function_async_method{this->peek().span()});
+      this->skip();
+      return this->parse_and_visit_function_parameters(v, name);
```

Comment:
Same comment as with `async export function`: we're not parsing the function body as if the function was `async`. But it's not easy here to fix that, so I think a TODO comment is fine.

Reply:
I'd like to work on that issue next if possible.